src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -542,6 +600,12 @@ private void CheckForUnobservedAwaitable(BoundExpression expression, DiagnosticB
 
         internal BoundStatement BindLocalDeclarationStatement(LocalDeclarationStatementSyntax node, DiagnosticBag diagnostics)
         {
+            if (IsEmbeddedStatementParent(node.Parent))
```

Comment:
>if (IsEmbeddedStatementParent(node.Parent)) [](start = 12, length = 43)

It feels like it would be better to intercept this statement in BindPossibleEmbeddedStatement function instead.